crates/biome_js_analyze/src/lint/nursery/use_explicit_type.rs

```diff
@@ -879,10 +927,8 @@ fn handle_any_function(func: &AnyJsFunction) -> Option<State> {
         return None;
     }
 
-    if is_function_used_in_argument_or_array(func) {
-        return None;
-    }
-
+    // TODO: why only arrow functions are ignored inside typed return?
```

Comment:
I'll point this out in a review comment as it really makes no sense to me. Why should `namedFunc` and `arrowFunc` be handled differently (`namedFunc` rejected, `arrowFunc` allowed) in the following case?

```ts
interface Behavior {
  attribute: string;
  namedFunc: () => string;
  arrowFunc: () => string;
}

function getObjectWithFunction(): Behavior {
  return {
    namedFunc: function myFunc() { return "value" },
    arrowFunc: () => {},
  }
};
```

Reply:
cc @concolas